src/goto-symex/goto_symex_state.cpp

```diff
@@ -389,8 +389,7 @@ bool goto_symex_statet::l2_thread_read_encoding(
   // see whether we are within an atomic section
   if(atomic_section_id!=0)
   {
-    guardt write_guard;
-    write_guard.add(false_exprt());
+    guardt write_guard = guardt(false_exprt());
```

Comment:
I suppose the compiler has to optimise that anyway, but I'd prefer `guardt write_guard(false_exprt());` instead of the assignment operator. (Applies here and below.)

Reply:
I personally prefer `guardt write_guard = guardt(false_exprt());` as `guardt write_guard(false_exprt());` looks a lot like a function declaration.

Reply:
I agree with @tautschnig.

Reply:
@tautschnig 
> How about guardt write_guard{false_exprt{}}; as the coding guidelines suggest anyway?

Yes I think that's best.